### DISCUSSION TITLE
Get call results after 1 minute

### DISCUSSION THREAD

**Discussion by Tilted-Software** [abuzuhri/Amazon-SP-API-CSharp#51] [Q&A]
Hi

Using the code below it hangs at the at the final line for precisely 1 minute:

            FikaAmazonAPI.Parameter.Order.ParameterOrderList serachOrderList = new FikaAmazonAPI.Parameter.Order.ParameterOrderList();
            serachOrderList.CreatedAfter = DateTime.UtcNow.AddMinutes(-600000);
            serachOrderList.OrderStatuses = new List<FikaAmazonAPI.Utils.Constants.OrderStatuses>();
            serachOrderList.OrderStatuses.Add(FikaAmazonAPI.Utils.Constants.OrderStatuses.Canceled);
            serachOrderList.MarketplaceIds = new List<string> { FikaAmazonAPI.Utils.MarketPlace.UnitedKingdom.ID };
            var orders = connection.Orders.GetOrders(serachOrderList);

Any reason for this?  Can it be stopped?

Thanks

Chris.

**Comment by abuzuhri** [comment]
Hello


Do you read usage limit ? 

https://github.com/abuzuhri/Amazon-SP-API-CSharp#usage-plans-and-rate-limits-in-the-selling-partner-api

**Reply by Tilted-Software** [reply]
Ah, got it - thanks for that.  So the rate limit on the orders call is 0.0167 giving a sleep time of 59880ms (1 minute) meaning it slept for this amount of time after making the call - this seems like a really long time!  I'm guessing the only way to reduce this would be to switch off the functionality and handle it myself?

**Reply by abuzuhri** [reply]
also you need to know this rate limit changed so you need to read from header .
If you manager your self who you can solve it ?

**Reply by Tilted-Software** [reply]
Fair enough, just doesn't seem practical to sleep for a minute after every call - especially where user interaction is required, it's always been like this with MWS too though so not really your issue, thanks again.
